src/actions/receive-profile.js

```diff
@@ -1072,9 +1073,7 @@ function _loadProbablyFailedDueToSafariLocalhostHTTPRestriction(
   return (
     error.name === 'TypeError' &&
     parsedUrl.protocol === 'http:' &&
-    (parsedUrl.hostname === 'localhost' ||
-      parsedUrl.hostname === '127.0.0.1' ||
-      parsedUrl.hostname === '::1') &&
+    localhostHostnames.includes(parsedUrl.hostname) &&
```

Comment:
nit: Would it make sense to add another function like isLocalURL that accepts a `URL`?

Reply:
actually I think we could make isLocalURL take both types of arguments, I'll update the PR with that.